gapii/cc/vulkan_inlines.inl

```diff
@@ -29,30 +31,30 @@ private:
 
 ////////// Command Buffer Commands
 template<>
-bool inline CommandListRecreator<std::shared_ptr<RecreateCmdUpdateBufferData>>::operator()(
+bool inline CommandListRecreator<std::shared_ptr<vkCmdUpdateBufferArgs>>::operator()(
```

Comment:
I'm sure its just my painkiller fueled state, but I'm confused. This code makes me think we're recreating the original vk commands from the `Args` data (presumably to encode them), but then we have something similar in gapis. Please can you describe in the CL description what this CL is actually doing?

Reply:
So the only thing that has actually CHANGED in vulkan_inlines is the name of the structures that we are encoding.
vkCmd*args instead of RecreateCmd...*.

MEC remains the same is always, except we now execute VkCmd* instead of RecreateCMD*, so we can remove all of the RecreateVkCmd* synthetic commands from the API file, and we no longer have to re-write all of the RecreateVkCmd* during state mutation.

And yes, we have identical logic in Go right now (and always) in order for us to cut apart command-buffers, we need to be able to generate a new command-buffer.

Once we only serialize state, all of this logic on the Interceptor side can go away, and will simpify significantly.

I will add some of this into the comment.